addon/services/router-scroll.js

```diff
@@ -10,23 +11,39 @@ export default Service.extend({
   }),
 
   scrollElement: 'window',
+  targetElement: null,
   delayScrollTop: false,
 
   init(...args) {
     this._super(...args);
     this._loadConfig();
-    set(this, 'scrollMap', {});
+    set(this, 'scrollMap', { default: { x: 0, y: 0 }});
     set(this, 'key', null);
   },
 
   update() {
     const scrollElement = get(this, 'scrollElement');
+    const targetElement = get(this, 'targetElement');
     const scrollMap = get(this, 'scrollMap');
     const key = get(this, 'key');
     let x;
     let y;
 
-    if ('window' === scrollElement) {
+    if (targetElement && '#' === targetElement.charAt(0)) {
```

Comment:
Do we need to use ids only here or can we use `document.querySelector` to find the element and support ids and classes?

Reply:
I think we should support both.  Changed.